src/app/pages/sign-in/sign-in.component.ts

```diff
@@ -80,10 +82,29 @@ export class SignInComponent implements OnInit {
       return;
     }
 
-    // Track which provider was used, will be needed once redirected back to Mauro
-    localStorage.setItem('openIdConnectProviderId', provider.id);
-    const redirectUrl = this.security.getOpenIdConnectAuthorizationUrl(provider);
-    window.open(redirectUrl.toString(), '_self');
+    // Try to synchronise the Mauro and SDE OpenID Connect providers to sequence in order
+    this.authenticationEndpoints
+      .listOauthProviders()
+      .pipe(
+        catchError(() => []),
+        map((sdeProviders) =>
+          // Big assumption - that Mauro and SDE have OpenID Connect providers are configured
```

Comment:
Can we use a more reliable detail than the label like the URI for the IDP?

Reply:
Unfortunately I don't think we can at the moment, it's a problem with storing the same Oauth provider details in two different systems. I've elaborated more in the comment instead.